src/interpreter/mod.rs

```diff
@@ -323,8 +323,11 @@ impl Interpreter {
     }
 
     pub fn set_prog_stmt_at(&mut self, index: usize, stmt: ast::Stmt) {
-        // Mark the stmt and everything following it for re-execution
-        self.pc = index;
+        // Mark the stmt and everything following it for re-execution,
+        // if we already executed that far
+        if self.pc > index {
+            self.pc = index;
+        }
```

Comment:
not that i'd think this is wrong, but is this better than `self.pc = min(self.pc, index);`?

Reply:
Just a matter of style